contributing.md

```diff
@@ -5,34 +5,39 @@ title: Contributing to Bazel
 
 # Contributing to Bazel
 
+There are many ways to help the Bazel project and ecosystem.
 
-## How can I help?
+## Provide feedback
 
-There are many ways to help the Bazel project and ecosystem.
+As you use Bazel, you may find things that can be improved. You can help by:
 
-1. Provide feedback as you use Bazel. As you use it, you may find things
-   that can be improved. Please report them:
-   - Report bugs. In particular, it's important to report any crash or
+   - Reporting bugs. In particular, it's important to report any crash or
      correctness bug.
-   - Report issues when the documentation is incomplete or unclear.
-   - Report issues when an error message could be improved.
+   - Reporting issues when the documentation is incomplete or unclear.
+   - Reporting issues when an error message could be improved.
+
+## Participate in the community
 
-1. As you get experienced with Bazel, engage with the community:
-   - Answer questions [on Stack Overflow](
+You can engage with the Bazel community by:
+
+   - Answering questions [on Stack Overflow](
      https://stackoverflow.com/questions/tagged/bazel).
-   - Help other users [on Slack](https://slack.bazel.build).
-   - Improve the documentation or [contribute examples](
+   - Helping other users [on Slack](https://slack.bazel.build).
+   - Improving documentation or [contributing examples](
      https://github.com/bazelbuild/examples).
-   - Share your experience or your tips, for example on a blog or social media.
+   - Sharing your experience or your tips, for example, on a blog or social media.
+
+## Contribute code
+
+You can contribute to the Bazel ecosystem by:
 
-1. Contribute to the Bazel ecosystem:
-   - Help rules maintainers by contributing pull requests.
-   - Create new rules and open-source them.
-   - Contribute to Bazel-related tools, for example migration tools.
-   - Improve Bazel integration with other IDEs and tools.
+   - Helping rules maintainers by contributing pull requests.
+   - Creating new rules and open-sourcing them.
+   - Contributing to Bazel-related tools, for example, migration tools.
+   - Improving Bazel integration with other IDEs and tools.
 
 
-## How can I contribute to Bazel source code?
+### How can I contribute to Bazel source code?
 
 Bazel is a large project and making a significant change to Bazel source code
```

Comment:
Bazel is a large project and making a significant change to Bazel source code can be difficult. Before making a change,  create a GitHub issue or email [bazel-dev@](https://groups.google.com/forum/#!forum/bazel-dev).

The most helpful contributions fix bugs or add features (as opposed to stylistic, refactoring, or "cleanup" changes). Your change should include tests and documentation, and keep in mind backward-compatibility,
portability, and the impact on memory usage and performance.

To learn how to work on the Bazel code base, see [getting started](basics/getting_started.html). To learn about how to submit a change, see the [patch acceptance process](basics/patching.html).

Reply:
Done.